### DISCUSSION TITLE
Setting the Default Browser (Possible yes, but how to manage in scoop)

### DISCUSSION THREAD

**Discussion by amreus** [ScoopInstaller/Scoop#4633] [Ideas]
This past week I have been experimenting with getting Windows to add scoop's `firefox` as a default browser. Windows does not allow programs to automatically set the default web browser - the user must intentionally pick the default browser from a list. However, I am happy to report that it is possible to add a scoop-installed browser to the list of default browsers. This is done simply by adding a few registry keys. Also good news is the registry keys are per-user in the `HKEY_CURRENT_USER` hive and do not require elevated permissions to modify. It is also possible to open the Default Browser Settings to encourage the user to select a different default browser.

So having established setting the default browser to a scoop-installed browser is easily possible, how should `scoop` manage browsers? Should each browser app manifest have its own `.reg` file? 

An idea I tried was registering a browser named "ScoopBrowser". When "ScoopBrowser" is selected as the default, it executes a scoop shim named `default-browser.exe`.  "ScoopBrowser" could be added to the registry just once during scoop's initial installation. By modifying `default-browser.shim` you can run any app without the need to write registry keys for each browser app installed. Of course in order for scoop to manage default browsers in this way, new functions would need to be added to `scoop`.

**Comment by rashil2000** [comment]
> Should each browser app manifest have its own `.reg` file?

Yes, that will be preferable.

> An idea I tried was registering a browser named "ScoopBrowser". When "ScoopBrowser" is selected as the default, it executes a scoop shim named `default-browser.exe`. "ScoopBrowser" could be added to the registry just once during scoop's initial installation. By modifying `default-browser.shim` you can run any app without the need to write registry keys for each browser app installed. Of course in order for scoop to manage default browsers in this way, new functions would need to be added to `scoop`.

The idea sounds great, but it's too specific a functionality to be added into Scoop core. I think something like can be very nicely achieved in a custom bucket (if someone wishes to make one - Scoop-Browsers), where one would install a one-time package for setting stuff up, and then install any browsers they want, which utilize/modify the mentioned `default-browser.shim` in their post_install step.

**Comment by amreus** [comment]
When selecting a default browser, should the name "scoop" appear as part of the browser name in the dialog (image below)? For example as "Firefox (Scoop)" instead of "Firefox"

I think yes in order to make it obvious that the user wants the scoop version of the browser and not some other possible installation of the same browser.


![2022-01-08_063929](https://user-images.githubusercontent.com/38442825/148643854-92a0275d-0561-4bbe-90c1-af7f009566c3.png)

**Reply by rashil2000** [reply]
> For example as "Firefox (Scoop)" instead of "Firefox"

Sure

**Comment by amreus** [comment]
Might anyone know if it's possible to disable Firefox's "Default Check" through scoop?  If the user uses Firefox's "Make Default" then it is going to register Firefox but it will no longer use the profile saved in the `persist` directory.

**Reply by rashil2000** [reply]
I found some links:
- http://forums.mozillazine.org/viewtopic.php?f=38&t=1486345
- http://kb.mozillazine.org/Locking_preferences
- https://www.askvg.com/tip-disable-default-browser-agent-after-installing-mozilla-firefox/ - Step 5

If these seem too difficult/flaky/invasive, we can just add a note in the manifest telling users to no change default browser settings.

**Comment by dajix350** [comment]
Is there any progress on this?

**Comment by niStee** [comment]
[[Proposal] Add to pwsh 'Open here' context menus (normal & privileged access) to Explorer
#1401 ](https://github.com/ScoopInstaller/Main/issues/1401)

**Comment by niStee** [comment]
#2651
